esp32/modules/splash.py

```diff
@@ -10,61 +10,99 @@
-    if sleeping:
-        info1 = 'Sleeping...'
-        info2 = 'Press any key to wake up'
+enableBpp = True
+requestedStandbyTime = 0
+
+def power_management(timeUntilNextTick):
+    global requestedStandbyTime
+    requestedStandbyTime = timeUntilNextTick
+    if timeUntilNextTick<0 or timeUntilNextTick>=60*5:
+        print("[PM] Next tick after more than 5 minutes (MAY BPP).")
+        global enableBpp
+        enableBpp = True
+        power_countdown_reset(0)
+    elif (timeUntilNextTick<=30):
+        print("[PM] Next loop in "+str(timeUntilNextTick)+" seconds (STAY AWAKE).")
+        power_countdown_reset(timeUntilNextTick)
+        return True #Service loop timer can be restarted
     else:
-        info1 = 'Press start to open the launcher'
-        if otaAvailable:
-            info2 = 'Press select to start OTA update'
-        else:
-            info2 = easyrtc.string(True, True)
-
-    l = ugfx.get_string_width(info1,"Roboto_Regular12")
-    ugfx.string(296-l, 0, info1, "Roboto_Regular12",ugfx.BLACK)
-    l = ugfx.get_string_width(info2,"Roboto_Regular12")
-    ugfx.string(296-l, 12, info2, "Roboto_Regular12",ugfx.BLACK)
-
-def splash_draw(full=False,sleeping=False):
+        print("[PM] Next loop in "+str(timeUntilNextTick)+" seconds (MAY SLEEP).")
+        global enableBpp
+        enableBpp = False
+        power_countdown_reset(0)
+    return False
 
-    vUsb = badge.usb_volt_sense()
-    vBatt = badge.battery_volt_sense()
-    vBatt += vDrop
-    charging = badge.battery_charge_status()
+powerCountdownOffset = 0
+def power_countdown_reset(timeUntilNextTick=-1):
+    global powerTimer
+    global powerCountdownOffset
+    try:
+        powerTimer.deinit()
+    except:
+        print("POWER TIMER DEINIT FAILED?!?!?")
+    if timeUntilNextTick>=0:
+        powerCountdownOffset = timeUntilNextTick
+    newTarget = powerCountdownOffset+badge.nvs_get_u8('splash', 'urt', 5) #User React Time (in seconds)
+    print("[SPLASH] Power countdown reset to "+str(newTarget))
+    powerTimer.init(period=newTarget*1000, mode=machine.Timer.ONE_SHOT, callback=power_countdown_callback)
+    
+def power_countdown_callback(tmr):
+    global enableBpp
+    draw(False, True)
+    services.force_draw(True)
+    draw(True, True)
+    global requestedStandbyTime
+    if requestedStandbyTime>0:
+        if enableBpp:
+            print("[PM] BPP for "+str(requestedStandbyTime)+" seconds.")
+            #appglue.start_bpp(round(requestedStandbyTime/60)) #BPP needs time in minutes
+            deepsleep.start_sleeping(requestedStandbyTime*1000)
+        else:
+            print("[PM] Sleep for "+str(round())+" seconds.")
+            deepsleep.start_sleeping(requestedStandbyTime*1000)
+    else:
+            print("[PM] BPP forever.")
+            #appglue.start_bpp(-1)
+            deepsleep.start_sleeping()
 
-    if splash_power_countdown_get()<1:
-        full= True
+# Graphics
 
-    if full:
+def draw(mode, goingToSleep=False):
+    if mode:
```

Comment:
I don't want the services.py to flush the framebuffer. This should be the responsibility of splash.py (or whatever hosts the services)

I agree that it isn't nice but it works so for now won't fix.